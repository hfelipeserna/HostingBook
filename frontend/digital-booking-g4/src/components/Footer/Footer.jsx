import React from "react";
import styles from "./Footer.module.css";

export default function Footer() {
  return (
    <footer className={styles.footerFijo}>
      <div className={styles.contenedorFooter}>
        <div className={styles.footerIzquierda}>
          <p>Â©2021 Hosting Book</p>
        </div>
        <div className={styles.footerDerecha}>
          <a
            className={styles.linkRedSocial}
            href="https://www.facebook.com/HostingBook-104939168676273"
            target="_blank"
            rel="noreferrer"
          >
            <i className="bx bxl-facebook-circle"></i>
          </a>
          <a
            className={styles.linkRedSocial}
            href="https://www.linkedin.com/in/hostingbook/"
            target="_blank"
            rel="noreferrer"
          >
            <i className="bx bxl-linkedin"></i>
          </a>
          <a
            className={styles.linkRedSocial}
            href="https://twitter.com/Hosting_Book"
            target="_blank"
            rel="noreferrer"
          >
            <i className="bx bxl-twitter"></i>
          </a>
          <a
            className={styles.linkRedSocial}
            href="https://www.instagram.com/hostingbook/"
            target="_blank"
            rel="noreferrer"
          >
            <i className="bx bxl-instagram-alt"></i>
          </a>
        </div>
      </div>
    </footer>
  );
}