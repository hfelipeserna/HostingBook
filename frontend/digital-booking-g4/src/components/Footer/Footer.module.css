.footerFijo{
    display: block;
    position: fixed;
    bottom: 0;
    left: 0;
    width: 100%;
    z-index: 999;
}

.contenedorFooter {
    display:flex;
    justify-content: space-between;
    align-items: center;
    background-color: var(--color-principal);
    padding: 15px 50px;
}

.footerIzquierda{
    color: var(--color-5);
    font-weight: 700;
    font-size: 0.875rem;
}

.footerDerecha {
    display:flex;
    justify-content: space-between;
    align-items: center;
    color: var(--color-5);
    
    font-weight: 700;
    font-size: 1.75rem;
    gap: 30px;
}

.footerDerecha i{
    cursor: pointer;
}

@media screen and (max-width:480px) {

    .contenedorFooter {
        padding: 20px;
    }

    .footerDerecha{
        display: none;
    }
    
}