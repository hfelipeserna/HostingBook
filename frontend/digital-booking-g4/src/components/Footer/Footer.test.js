import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { createSerializer } from "enzyme-to-json";
import Footer from "./Footer";

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

describe("Footer tests", function () {
  let footer;

  beforeEach(() => {
    footer = shallow(<Footer />);
  });

  test("Devuelve el footer correctamente", () => {
    expect(footer).toMatchSnapshot();
  });

  test("El footer contiene 1 p con el copyright", () => {
    let node = footer.find("p");
    expect(node.length).toEqual(1);
  });

  test("El footer contiene 4 iconos con las redes sociales", () => {
    let node = footer.find("i");
    expect(node.length).toEqual(4);
  });
});
