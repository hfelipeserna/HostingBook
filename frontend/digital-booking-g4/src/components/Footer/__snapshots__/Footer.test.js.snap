// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer tests Devuelve el footer correctamente 1`] = `
<footer
  className="footerFijo"
>
  <div
    className="contenedorFooter"
  >
    <div
      className="footerIzquierda"
    >
      <p>
        Â©2021 Hosting Book
      </p>
    </div>
    <div
      className="footerDerecha"
    >
      <a
        className="linkRedSocial"
        href="https://www.facebook.com/HostingBook-104939168676273"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="bx bxl-facebook-circle"
        />
      </a>
      <a
        className="linkRedSocial"
        href="https://www.linkedin.com/in/hostingbook/"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="bx bxl-linkedin"
        />
      </a>
      <a
        className="linkRedSocial"
        href="https://twitter.com/Hosting_Book"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="bx bxl-twitter"
        />
      </a>
      <a
        className="linkRedSocial"
        href="https://www.instagram.com/hostingbook/"
        rel="noreferrer"
        target="_blank"
      >
        <i
          className="bx bxl-instagram-alt"
        />
      </a>
    </div>
  </div>
</footer>
`;
